starter-project-backend-g32/src/controllers/user.controller.ts

```diff
@@ -0,0 +1,75 @@
+import express, {Request, Response, NextFunction} from 'express'
+import userModel from '../models/user.models';
+
+
+export const getSpecificUser = async (req: Request, res: Response) => {
+    try {
+      const doc = await userModel.findOne({ _id: req.params.id }).lean().exec();
+
+      if (!doc) {
+        return res.status(404).end();
+      }
+
+      res.status(200).json({ data: doc });
+    } catch (e) {
+      console.error(e);
+      res.status(404).end();
+    }
+};
+export const getAllUsers = async (req: Request, res: Response) => {
+    try {
+      const docs = await userModel.find().lean().exec();
+      res.status(200).json({ data: docs });
+    } catch (e) {
+      console.error(e);
+      res.status(404).end();
+    }
+};
+export const createUser = async (req: Request, res: Response) => {
+    try {
+      const doc = await userModel.create({ ...req.body });
```

Comment:
passing the whole req.body is not recommended so better to extract each property from req.body and pass it to create function